runtime/ts/storage/storage-provider-factory.ts

```diff
@@ -35,7 +35,10 @@ export class StorageProviderFactory {
     return instance;
   }
 
-  _storageForKey(key) {
+  _storageForKey(key: string) {
+    if (!key) {
```

Comment:
Isn't this kind of thing what we normally use `assert`s for? I could well be wrong, it's not a rhetorical question...

Reply:
My rule is generally throw Errors for user-supplied inputs, use asserts for testing internal state.  Others might disagree?